### PR TITLE
 Reflect recent Promises/A+ support from Mojo::Pg into Mojo::mysql

### PR DESCRIPTION
**Pull request by mikemagowan**
PR is based on merging in changes from Mojo::Pg.  Specifically:

- [https://github.com/kraih/mojo-pg/commit/288faf0005910cdbad44cd4bcce7c4c2afbe9a08](https://github.com/kraih/mojo-pg/commit/288faf0005910cdbad44cd4bcce7c4c2afbe9a08)
- [https://github.com/kraih/mojo-pg/commit/7cd5ff262daf9c09b6f46ef08da816fd4fabf9d7](https://github.com/kraih/mojo-pg/commit/7cd5ff262daf9c09b6f46ef08da816fd4fabf9d7)
- [https://github.com/kraih/mojo-pg/commit/312ef4758041d9269c33cda9ea8a56c2500ca66b](https://github.com/kraih/mojo-pg/commit/312ef4758041d9269c33cda9ea8a56c2500ca66b)
- [https://github.com/kraih/mojo-pg/commit/d9c467f3217634da82ff2e64f1ce908061cf4bfa](https://github.com/kraih/mojo-pg/commit/d9c467f3217634da82ff2e64f1ce908061cf4bfa)
- [https://github.com/kraih/mojo-pg/commit/a4210860a57d3db0c4c04dfbb66be52d54763bf6](https://github.com/kraih/mojo-pg/commit/a4210860a57d3db0c4c04dfbb66be52d54763bf6)

Additionally the tests have been tweaked to make them work without some of the Pg features (namely: returning).

Finally a fix had to be applied to .travis.yml to change the default database collation to be utf8 as it seemed to be defaulting to latin1 causing significant test failure.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
 sudo: false
 perl:
+  - "5.26"
+  - "5.24"
+  - "5.22"
   - "5.20"
+  - "5.18"
   - "5.16"
+  - "5.14"
+  - "5.12"
```
Comment (jhthorsen) [inline .travis.yml:11]: Why do you need so many perl versions?

**Comment by jhthorsen** [comment]
Looks great! Thanks for your contribution 👍

File: .travis.yml (hunk 1 above)
Comment (mikemagowan) [reply .travis.yml:11]: Was reflecting the same list from Mojo::Pg.

Comment (Grinnz) [reply .travis.yml:11]: Seems reasonable to include all the standard ones that are supported IMO

Comment (jhthorsen) [reply .travis.yml:11]: I see. Thank you.

**Pull request merged**